examples/mem-alloc-test/src/main.rs

```diff
@@ -0,0 +1,293 @@
+        }
+        MemSize::Medium => {
+            data.avg_duration_medium_thread += avg_thread_latency;
+        }
+        MemSize::Small => {
+            data.avg_duration_small_thread += avg_thread_latency;
+        }
+    };
+}
+
+fn get_random_num(start: usize, end: usize) -> usize {
+    let mut rng = rand::thread_rng();
+    rng.gen_range(start..=end)
+}
+
+fn wait_per_thread(pair_clone: Arc<(Mutex<bool>, Condvar)>) {
+    let (lock, cvar) = &*pair_clone;
+    let mut started = lock.lock().unwrap();
+    while !*started {
+        started = cvar.wait(started).unwrap();
+    }
+    drop(started);
+}
+
+fn wakeup_all_child_threads(pair_clone: Arc<(Mutex<bool>, Condvar)>) {
+    let (lock, cvar) = &*pair_clone;
+    let mut started = lock.lock().unwrap();
+    *started = true;
+    cvar.notify_all();
+    drop(started);
+}
+
+fn traverse_buffer(buf: &mut Vec<i32>, scan_interval: usize) {
+    let mut ptr = 0;
+    let num_indices_checks = get_random_num(1, MAX_INDEX_CHECKS_PER_BUFFER);
+    for i in 1..=num_indices_checks {
+        /* Check for random indices and number of such indices is  num_indices_checks
+         * Please note that depending on the number random number generator, we
+         * can check for the same index multiple times. We could have checked
+         * for all the indices but that would be too time consuming
+         */
+        let index = get_random_num(0, buf.len() - 1);
+        buf[index] = buf[index] * 2;
+    }
+}
+
+fn worker_thread(
+    tid: i32,
+    shared_mutex_clone: Arc<Mutex<Counters>>,
+    memsize: MemSize,
+    pair_clone: Arc<(Mutex<bool>, Condvar)>,
+) {
+    /* Wait for all the threads to be created and then start together */
+    wait_per_thread(pair_clone);
+
+    let mut count = 0;
+    let mut tot_time_ns = 0;
+
+    /* Once the thread's allocation and deallocation operations begin, we
+     * shouldn't take any lock as the allocator that we trying to test is a
+     * multithreaded allocator and we should allow as many threads as possible
+     * to get the lock.
+     */
+    loop {
+        /* Create a random size depending on the memory type */
+        let (scan_interval, size, limit) = match memsize {
+            MemSize::Large => {
+                /* buffer size will be from 1GB to 4GB */
+                (
+                    SCAN_INTERVAL_LARGE_THREAD,
+                    TO_GB * get_random_num(LARGE_THREAD_MEM_START, LARGE_THREAD_MEM_END),
+                    LIMIT_LARGE_THREAD,
+                )
+            }
+            MemSize::Medium => {
+                /* buffer size will be from 8MB to 128 */
+                (
+                    SCAN_INTERVAL_MEDIUM_THREAD,
+                    TO_MB * get_random_num(MEDIUM_THREAD_MEM_START, MEDIUM_THREAD_MEM_END),
+                    LIMIT_MEDIUM_THREAD,
+                )
+            }
+            MemSize::Small => {
+                /* buffer size will be from 1KB to 512KB */
```

Comment:
For a heap allocator 1KB is still big. Can this start from 1B instead? Corner cases may only pop up with such tiny chunks